### PR TITLE
Bump to version Docker 18.09

### PR DESCRIPTION
**Pull request by dims**
Kubernetes master is now at 18.09:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/system/docker_validator.go#L42

It was updated for 1.14 release:
https://github.com/kubernetes/kubernetes/commit/62fc6df5c2c6e31c8ec15109408c72b6e1dbc6d2

Change-Id: Iab02ca4ef0de0a11fc763ef36418b85d77007220

### CODE REVIEW

**Comment by dims** [comment]
```
root@kind-control-plane:/# crictl version
Version:  0.1.0
RuntimeName:  docker
RuntimeVersion:  18.09.4
RuntimeApiVersion:  1.39.0
root@kind-control-plane:/# crictl -r /run/containerd/containerd.sock version
Version:  0.1.0
RuntimeName:  containerd
RuntimeVersion:  1.2.5
RuntimeApiVersion:  v1alpha2
```

File: images/base/Dockerfile [hunk 1]
```diff
@@ -62,9 +62,9 @@ RUN clean-install \
 # - add the fingerprint
 # - add the repository
 # - update apt, install docker, cleanup
-# NOTE: 18.06 is officially supported by Kubernetes currently, so we pin to that.
+# NOTE: 18.09 is officially supported by Kubernetes currently, so we pin to that.
 # https://kubernetes.io/docs/tasks/tools/install-kubeadm/
-ARG DOCKER_VERSION="18.06.*"
+ARG DOCKER_VERSION="5:18.09.*"
```
Comment (BenTheElder) [inline images/base/Dockerfile]: is this `5:` intentional? what does it do?

File: images/base/Dockerfile [hunk 2]
```diff
@@ -75,6 +75,10 @@ RUN curl -fsSL "https://download.docker.com/linux/$(. /etc/os-release; echo "$ID
         "deb https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") $(lsb_release -cs) stable" \
     && clean-install "docker-ce=${DOCKER_VERSION}"
 
+# Enable CRI for containerd
+RUN mkdir -p /etc/docker && echo '{"cri-containerd": true}' > /etc/docker/daemon.json
```
Comment (BenTheElder) [inline images/base/Dockerfile]: FYI @Random-Liu @yujuhong -  🎉 https://github.com/kubernetes-sigs/kind/pull/425#issuecomment-479168589

File: images/base/Dockerfile (hunk 1 above)
Comment (dims) [reply images/base/Dockerfile]: the version is `Version: 5:18.09.4~3-0~ubuntu-bionic` for some reason in the packages... 18.06 did not have a prefix

```
root@kind-control-plane:/# apt-cache madison docker-ce
 docker-ce | 5:18.09.4~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.3~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.2~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.1~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.0~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.3~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.2~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.1~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.0~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.03.1~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
```

Comment (neolit123) [reply images/base/Dockerfile]: https://readme.phys.ethz.ch/documentation/debian_version_numbers/
> Package with epoch, e.g. if upstream changed from date based to major.minor versioning
Old upstream version: 2013.06.06
New upstream version: 1.0
Problem: 1.0 < 2013.06.06
Solution: Add a (debian-specific) epoch:
Old package version (example): 2013.06.06-4 (upstream version + debian revision + implicit epoch of "0", i.e. 0:2013.06.06 is equivalent)
New package version: 1:1.0-1 (epoch of 1 + upstream version + debian revision)

o_O

Comment (BenTheElder) [reply images/base/Dockerfile]: interesting @neolit123 @dims, thanks.

**Comment by BenTheElder** [comment]
this looks good, thank you! I will add a commit to update to a base image built with it later.

**Pull request merged**